sport/app/football/feed/agent.scala

```diff
@@ -154,10 +154,30 @@ class CompetitionAgent(
         }
       }
 
-      //it is important that newMatches are at the start of the list here
-      comp.copy(matches = (newMatches ++ comp.matches).sorted(MatchStatusOrdering).distinctBy(_.id).sortByDate)
+      val allMatches = (newMatches ++ comp.matches).sorted(MatchStatusOrdering).distinctBy(_.id).sortByDate
+
+      val updatedMatches = allMatches.filter(footballMatch =>
+        (!isPlaceholderMatch(footballMatch)) ||
+          (isPlaceholderMatch(footballMatch) && !nonPlaceHolderMatchWithSameDateExists(allMatches, footballMatch)),
+      )
+
+      comp.copy(matches = updatedMatches)
     }
 
+  def isPlaceholderMatch(footballMatch: FootballMatch): Boolean = {
```

Comment:
scala thought - this might be worth making `private` if we want to keep it so we don't leak this method as it's very specific to this issue.

Reply:
I would like to keep this for now as there is a unit test for this. I will revisit tomorrow to re-write the tests and make this `private`